### ISSUE TITLE
Enhance background colour cycling in editor

### ISSUE THREAD

**Issue by mzuther** [Beep6581/RawTherapee#4027]
Hi!

I currently use RawTherapee 5.2 (stable) on Linux Mint.

To check whether the edited image is too light or too dark, I regularly cycle the editor's background colours (using the key command "9"). The theme-based background with transparency is great for cropping images, but I also need a opaque grey background to check for tinted colours. And changing this in the settings is too much hassle to regularly do it.

I propose two use enhancements to choose from:

1. Separate colour cycling and transparency: use "9" for cycling the background colour and, say, "8" for toggling transparency. This means that one could use transparency with *any* background colour and have a neutral grey with and without transparency. My preferred solution, but a little harder to implement.

2. Add a neutral grey to the colour cycling: "9" would cycle through theme-based (with transparency), white, neutral grey (#7c7c7c or such) and black. Easy. :)

Thanks for putting so much energy into RawTherapee - and for listening!

Martin

**Comment by Beep6581** [comment]
Hi @mzuther 

I agree that it would be convenient to be able to easily set the canvas/crop mask color to average grey. The first proposed solution is overkill. Simplicity is best. Adding a fourth mode, "neutral gray", is simple and would fulfill my needs.

**Comment by Beep6581** [comment]
Patch ready, icons not yet and I have to head out, I'll PR tomorrow.

**Comment by mzuther** [comment]
Good heavens, you're fast! :)

**Comment by Beep6581** [comment]
https://github.com/Beep6581/RawTherapee/compare/bgcolor-vbox

![imgur-2017_08_20-19 34 33](https://user-images.githubusercontent.com/5844619/29496966-acfb922c-85de-11e7-9ef2-204866eca8d7.png)

I'm not fond of those vertical buttons:
1. They do the same job as their red/green/blue/L/focusmask neighbors to the right, so there is no good reason for them to look different.
2. The images those buttons contain are made of a 1px high horizontal line surrounded by a 1px frame, making them 3px in height. It's not clear what color this 1px line is, it's just too small to see on a full HD screen, let alone HiDPI. i.e. the frame doesn't help.
3. These images were not consistently implemented across the 3 buttons regarding color and alpha.
4. All three buttons have an on and off image, but of those three sets of images only two actually differ between the on and off state (`previewmodeBC0-off.png` is identical to `previewmodeBC0-on.png`).

So I propose [bgcolor-h](https://github.com/Beep6581/RawTherapee/pull/4029) instead. The buttons are now consistent with their semantic counterparts and are clear to see regardless of screen resolution.

![imgur-2017_08_20-20 33 29](https://user-images.githubusercontent.com/5844619/29497372-ea768334-85e6-11e7-8337-c8383d5aaea8.png)
![imgur-2017_08_20-19 37 39](https://user-images.githubusercontent.com/5844619/29497373-ea77030e-85e6-11e7-89aa-37f6192879d2.png)

**Comment by mzuther** [comment]
I see the point. But I never found the colour boxes nice - they seem to fall out of place because of their small size. The new large grey-with-transparency button doesn't really help in this regard (please don't take this as criticism!).

What about enlarging all the buttons like here? I think it looks quite nice...

![improved_buttons](https://user-images.githubusercontent.com/316423/29497789-d1a3ae06-85ee-11e7-99f3-2e192c95734f.png)

**Comment by Hombre57** [comment]
What I'm not fond of light frame for r, g, b channel and light background. Could they be darker than their content ? Otherwise 👍 for @Beep6581 or @mzuther version.

**Comment by Beep6581** [comment]
Note to self: merge PR #4030 when merging PR #4029

**Comment by Beep6581** [comment]
To experiment:
```css
#EditorTopPanel .image-button.toggle:checked {
	background: transparent;
	border: 1px solid transparent;
}
```

@Hombre57 I experimented with 1 and 2 pixel wide borders, trying lighter and darker color combinations and transparencies, but they all looked bad. In the end a simple solid color with or without a 1px black border looked best across all themes. As two of the buttons require a border for some themes, they all have a border.

![imgur-2017_08_26-19 21 52](https://user-images.githubusercontent.com/5844619/29743568-65dd29bc-8a95-11e7-8a9f-b2a796808ea8.png)

The focus-screen button, while looking nice, did not match the other buttons, so I replaced it too:
![imgur-2017_08_26-19 29 21](https://user-images.githubusercontent.com/5844619/29743567-65dccf1c-8a95-11e7-9729-95916ec641b1.png)

Any objections to commit?

**Comment by mzuther** [comment]
I just compared the last picture you posted to the first one. This looks *much* better than before! If you ask me, go ahead... :)

**Comment by Hombre57** [comment]
@Beep6581 Way much better, you're right ! However I'd keep the big version for the focus mask, and move it to the right of the separator. That way we would have the channels in a "section" and the alert buttons (focus, HL, Shadows) in another "section". That would be more logical to me. Then yes, you can commit.

**Comment by Beep6581** [comment]
@Hombre57 could you review and comment whether its ok to merge?

**Comment by Hombre57** [comment]
@Beep6581 Once you've fixed the capitalization in language strings identifier, I'm fine with committing. It looks really nicer and consistent now 👍

**Comment by Beep6581** [comment]
Branch `bgcolor-h` and @TooWaBoo 's theme fix branch merged.

The new features will be available in RawTherapee 5.3, out soon.

**Comment by mzuther** [comment]
Thanks, it works great! :)
